packages/cli/src/commands/info/info_command.ts

```diff
@@ -0,0 +1,51 @@
+import * as os from 'node:os';
+import { Argv, CommandModule } from 'yargs';
+import {
+  formatCdkInfo,
+  formatEnvInfo,
+  getCdkInfo,
+  getEnvInfo,
+} from '../../info/index.js';
+import { LogLevel, Printer } from '@aws-amplify/cli-core';
+
+/**
+ * Represents the InfoCommand class.
+ */
+export class InfoCommand implements CommandModule<object> {
+  /**
+   * @inheritDoc
+   */
+  readonly command: string;
+
+  /**
+   * @inheritDoc
+   */
+  readonly describe: string;
+
+  /**
+   * Creates top level entry point for generate command.
+   */
+  constructor() {
+    this.command = 'info';
+    this.describe = 'Generates information for Amplify troubleshooting';
+  }
+
+  /**
+   * @inheritDoc
+   */
+  handler = async (): Promise<void> => {
+    const printer = new Printer(LogLevel.INFO);
+    const envInfo = await getEnvInfo();
+    const formattedEnvInfo: string = formatEnvInfo(envInfo);
+    const cdkInfo = await getCdkInfo();
+    const formattedCdkInfo: string = formatCdkInfo(cdkInfo);
+    printer.print(`${formattedEnvInfo}${os.EOL}${formattedCdkInfo}`);
+  };
```

Comment:
Please refactor this in such a way that:
1. Classes/types are created and injected into command for testability. See how `ConfigureProfileCommand` is implemented for example. Some ideas `class LocalEnvironmentDetailsReader`, `type LocalEnvironmentDetails { environment: {}, cdk: {}`, `class LocalEnvironmentDetailsFormatter`.
2. Please avoid acronyms when possible. I.e. CDK is fine, but `environment` could be better.
3. Printer should also be injected, there should be an instance somewhere. (Look around in cli package to see the pattern).

Reply:
Specifically, [this printer instance](https://github.com/aws-amplify/amplify-backend/blob/main/packages/cli/src/printer.ts) should be used